edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/brokerConnection/MqttBrokerConnector.cs

```diff
@@ -189,7 +194,21 @@ public async Task<bool> SendAsync(string topic, byte[] payload, bool retain = fa
                 new Exception("Could not store message id to monitor Mqtt ACK");
             }
 
-            var result = await tcs.Task.TimeoutAfter(TimeSpan.FromSeconds(AckTimeoutSecs));
+            bool result;
+
+            try
+            {
+                result = await tcs.Task.TimeoutAfter(this.AckTimeout);
+            }
+            catch
+            {
+                lock (this.guard)
```

Comment:
why not use `ConcurrentDictionary`?

Reply:
There is a place when need to keep the dictionary locked until another operation is finished, so that is why I use a regular lock. Because of that, every other place needs to be locked, so at the end (because of the manual locks) a ConcurrentDictionary would not give any benefit.